### PR TITLE
'Clear()' image operation

### LINKED ISSUE #41: Add Clear() methods to force drawing the provided color over a region
This is a special case of clear with a forced `PixelAlphaCompositionMode.Src` & `PixelColorBlendingMode.Normal`.

For best efficiency this is dependent on #40 so you can composite the fastest path without have to special case to many things.

### PR DESCRIPTION
**Pull request by tocsoft**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Drawing/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Adds `Clear(..)` apis that will force the brush to overwrite the color of the destination image.

For most efficiency this works with #42 to create a fast path for most of the clear operations.

Fixes #41

Dependent on #39

### CODE REVIEW

File: src/ImageSharp.Drawing/Processing/Extensions/ClearExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+    /// <summary>
+    /// Adds extensions that allow the clearing of regions with various brushes to the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static class ClearExtensions
+    {
+        /// <summary>
+        /// Clones the graphicoptions and applies changes required to force clearing.
+        /// </summary>
+        /// <param name="options">The optinos to clone</param>
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearExtensions.cs:17]:
```suggestion
        /// <param name="options">The options to clone.</param>
```

File: src/ImageSharp.Drawing/Processing/Extensions/ClearExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+namespace SixLabors.ImageSharp.Drawing.Processing
+{
+    /// <summary>
+    /// Adds extensions that allow the clearing of regions with various brushes to the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static class ClearExtensions
+    {
+        /// <summary>
+        /// Clones the graphicoptions and applies changes required to force clearing.
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearExtensions.cs:15]:
```suggestion
        /// Clones the graphics options and applies changes required to force clearing.
```

File: src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+using SixLabors.ImageSharp.Drawing.Processing.Processors.Drawing;
+using SixLabors.ImageSharp.Processing;
+
+namespace SixLabors.ImageSharp.Drawing.Processing
+{
+    /// <summary>
+    /// Adds extensions that allow the filling of polygon outlines to the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static class ClearPathExtensions
+    {/// <summary>
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs:13]: Formatting has gone awry here.

File: src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     /// Clones the graphicoptions and applies changes required to force clearing.
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs:14]:
```suggestion
     /// Clones the graphics options and applies changes required to force clearing.
```

File: src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+     /// </summary>
+     /// <param name="shapeOptions">The optinos to clone</param>
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs:16]:
```suggestion
     /// <param name="shapeOptions">The options to clone.</param>
```

File: src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        /// <summary>
+        /// Flood fills the image in the shape of the provided polygon with the specified brush without any blending.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="options">The graphics options.</param>
+        /// <param name="brush">The brush.</param>
+        /// <param name="path">The shape.</param>
+        /// <returns>The <see cref="Image{TPixel}"/>.</returns>
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs:35]: The return type here and in other places. We should fix this as it breaks the API docs.

Comment (tocsoft) [reply src/ImageSharp.Drawing/Processing/Extensions/ClearPathExtensions.cs:35]: fixed all in the repo

**Pull request merged**